frontend/src/scenes/insights/InsightPageHeader.tsx

```diff
@@ -242,9 +242,11 @@ export function InsightPageHeader({ insightLogicProps }: { insightLogicProps: In
                                             label="Debug panel"
                                         />
                                     ) : null}
-                                    {hogQL && (
-                                        <>
-                                            <LemonDivider />
+
+                                    {(hogQL || showCohortButton) && <LemonDivider />}
+                                    {hogQL &&
+                                        !isHogQLQuery(query) &&
+                                        !(isDataVisualizationNode(query) && isHogQLQuery(query.source)) && (
```

Comment:
think you don't need this line since you've already checked `!isHogQLQuery(query.source)` on line 248

`!a && !(b && a) == !a`

Reply:
No this checks for the case when it's a data visualisation node that has  hogql as source (as opposed to the top level query node being hogql)